### PR TITLE
feat(data): update data format

### PR DESCRIPTION
**Pull request by lionelB**
use postgrest endpoint with resources embedding to made only one http request
add types checking

breaking change: agreement is removed from references item
- add new dila* key in references
- add internal id for question and answer

### CODE REVIEW

**Comment by lionelB** [comment]
## avant 

```json
{
          "markdown": "Si un poste se libère ou est créé dans l'entreprise, l'employeur doit en informer les salariés. Il étudie en priorité les candidatures internes répondant aux conditions du poste.",
          "idcc": "0029",
          "references": [
            {
              "category": "agreement",
              "title": "Article 04.05.1",
              "agreement": {
                "etat": "VIGUEUR_NON_ETEN",
                "id": "KALICONT000005635234",
                "nature": "IDCC",
                "num": 29,
                "shortTitle": "Hospitalisation privée : établissements privés d'hospitalisation, de soins, de cure et de garde à but non lucratif (FEHAP)",
                "texte_de_base": "KALITEXT000029787837",
                "title": "Convention collective nationale des etablissements privés d'hospitalisation, de soins, de cure et de garde à but non lucratif du 31 octobre 1951.",
                "url": "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635234"
              }
            }
          ]
        }
```


## apres

```json
        {
          "id": "a31de674-b288-479c-94e4-42ad52eda030",
          "idcc": "0029",
          "markdown": "Si un poste se libère ou est créé dans l'entreprise, l'employeur doit en informer les salariés. Il étudie en priorité les candidatures internes répondant aux conditions du poste.",
          "references": [
            {
              "title": "Article 04.05.1",
              "url": "https://beta.legifrance.gouv.fr/conv_coll/id/KALIARTI000029952604/?idConteneur=KALICONT000005635234",
              "dila_id": "KALIARTI000029952604",
              "dila_cid": "KALIARTI000005787080"
            }
          ]
        },
```

File: src/fetchContributions.js [hunk 1]
```diff
@@ -20,104 +66,37 @@ const sortByKey = (key) => (a, b) => `${a[key]}`.localeCompare(`${b[key]}`);
...
-  const getGenericAnswer = (questionId) => {
-    const genericAnswer = answers.find(
-      (a) => a.question_id === questionId && a.agreement_id === null
-    );
-    if (!genericAnswer) {
-      return;
+  return /**@type {Contributions.Question[]} */ (questions.flatMap(
+    ({ id, index, value: title, answers }) => {
+      const genericAnswer = getGenericAnswer(answers);
+      if (!genericAnswer) return [];
```
Comment (revolunet) [inline src/fetchContributions.js:80]: `array` when no genericAnswer, `object` otherwise ?

File: src/fetchContributions.js [hunk 2]
```diff
@@ -1,17 +1,63 @@
...
+}
+
+/** @type {Contributions.createRefFn} */
+const getRefUrl = (idcc) => (reference) => {
+  const agreement = kaliData.find(
+    (convention) => comparableIdcc(convention.num) === comparableIdcc(idcc)
+  );
+  if (!agreement) {
+    return reference;
+  }
```
Comment (revolunet) [inline src/fetchContributions.js]:
```suggestion
```

File: src/fetchContributions.js (hunk 1 above)
Comment (lionelB) [reply src/fetchContributions.js:80]: returning '[]' in flatMap will remove the entry, so no need to filter question without genericAnswer later

File: src/fetchContributions.js (hunk 2 above)
Comment (revolunet) [reply src/fetchContributions.js]: nop

**Review by revolunet — APPROVED** [review]

File: src/fetchContributions.js (hunk 2 above)
Comment (UnbearableBear) [reply src/fetchContributions.js]: Héhé, nop mais il y a effectivement moyen de ne pas avoir de double return:

```
if(agreement) {
  reference.url = reference.dila_id 
    ? `https://beta.legifrance.gouv.fr/conv_coll/id/${reference.dila_id}/?idConteneur=${agreement.id}`
    : agreement.url;
}
```

**Pull request merged**